### PR TITLE
Detect user OS and open config file with correct text editor

### LINKED ISSUE #20: Attempts to open newly created config with vim
The current version of the tool defaults to opening the created config file with vim.

This can result in an error of the config not opening if vim is not installed.

Proposed changes:

- Force user to install vim
- Detect user OS (if Windows: open with notepad)

The code causing the error:
`os.system("vim {}".format(CONFIG_FILE_PATH))`

### PR DESCRIPTION
**Pull request by KevinLu**
Needs testing on Linux and MacOS before we can merge into master branch.

Feel free to change the method for file opening or change the code styling.

This PR resolves #20

### CODE REVIEW

File: fund_my_watcard/watcard [hunk 1]
```diff
@@ -84,7 +84,7 @@ def reset_config_file():
     report_message("Reset config file successfully. Try 'watcard --config' to fill your information.")
 
 
-def write_template_to_config_file():
+def write_template_to_config_file):
```
Comment (KevinLu) [inline fund_my_watcard/watcard]: I think this is missing a left bracket so it is causing code to fail.

**Comment by xingweitian** [comment]
I have already done some tiny tweaks, could you please test again whether Windows will be happy with this change?

**Comment by xingweitian** [comment]
By the way, I have tested in macOS, it works properly.

**Comment by KevinLu** [comment]
I tested on Windows 10 and it works perfectly.

**Pull request merged**